package/piksi_system_daemon/src/main.c

```diff
@@ -347,22 +347,60 @@ static void img_tbl_settings_setup(settings_ctx_t *settings_ctx)
   }
 }
 
-static void hardware_variant_settings_setup(settings_ctx_t *settings_ctx)
+static void hardware_info_settings_setup(settings_ctx_t *settings_ctx)
 {
-  static char *hardware_variant = "";
-  if (device_is_duro()) {
-    hardware_variant = "Duro";
-  } else {
-    hardware_variant = "Multi";
+  static char info_string[STR_BUFFER_SIZE] = {0};
+
+  if (hw_version_string_get(info_string, sizeof(info_string)) != 0) {
+    piksi_log(LOG_WARNING, "Failed to get hw_version for system_info registration");
+  }
+  else if (settings_register_readonly(settings_ctx,
+                                      "system_info",
+                                      "hw_version",
+                                      info_string,
+                                      strlen(info_string) + 1,
+                                      SETTINGS_TYPE_STRING)
+           != 0) {
+    piksi_log(LOG_WARNING, "Failed to register hw_version in system_info");
+  }
+
+  if (hw_revision_string_get(info_string, sizeof(info_string)) != 0) {
+    piksi_log(LOG_WARNING, "Failed to get hw_revision for system_info registration");
+  }
+  else if (settings_register_readonly(settings_ctx,
+                                      "system_info",
+                                      "hw_revision",
+                                      info_string,
+                                      strlen(info_string) + 1,
+                                      SETTINGS_TYPE_STRING)
+           != 0) {
+    piksi_log(LOG_WARNING, "Failed to register hw_revision in system_info");
+  }
+
+  if (hw_variant_string_get(info_string, sizeof(info_string)) != 0) {
+    piksi_log(LOG_WARNING, "Failed to get hw_variant for system_info registration");
+  }
+  else if (settings_register_readonly(settings_ctx,
+                                      "system_info",
+                                      "hw_variant",
+                                      info_string,
+                                      strlen(info_string) + 1,
+                                      SETTINGS_TYPE_STRING)
+           != 0) {
```

Comment:
For these register calls, is it not necessary to provide a unique pointer for each string?

Reply:
I agree it's a bit lazy but technically for read only settings it's only necessary to populate the variable pointer with the right value for the initial registration which happens serially and won't change at or be referenced after that (read requests are serviced by the `sbp_settings_daemon`)